lib/first_run.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * @fileOverview This file opens a tab with the first run html doc under
+ * appropriate conditions when the extension is loaded on browser launch or
+ * on extension install.
+ *
+ * Appropriate conditions fall under two circumstances:
+ * 1. LocalStorage does not have a stored value for the version of privly
+ *    installed. (Privly was just installed or localStorage was cleared)
+ * 2. The version stored in the manifest differs from the version stored in
+ *    localStorage. (Privly was updated)
+ *
+ **/
+
+// Local storage shim
+var { ls } = require("./local_storage.js");
+
+/**
+ * @namespace for the firstRun functionality.
+ */
+var firstRun = {
+  /**
+   * Get version stored in the manifest
+   */
+  getPrivlyVersion: function() {
+    return require("sdk/self").version;
+  },
+
+  /**
+   * Get version stored in localStorage
+   */
+  getStoredVersion: function() {
+    var stored_version = ls.getItem("version");
+    return stored_version;
+  },
+
+  /**
+   * Update localStorage version
+   */
+  updateVersion: function(version) {
+    ls.setItem("version", version);
+  },
+
+  /**
+   * Check whether the first run html page should be opened.
+   */
+  runFirstRun: function() {
+
+    // Initialize the spoofing glyph
+    // The generated string is not cryptographically secure and should not be used
+    // for anything other than the glyph.
+    if (ls.getItem("glyph_cells") === void(0)) {
```

Comment:
I read it's better to use void(0) instead of the error-prone undefined because it can be redefined.